### ISSUE TITLE
inconsistency in the scaling of the y-axes

### ISSUE THREAD

**Issue by michkami** [naver/billboard.js#3691]
## Description
In the example below we have 3 data rows. 
**Data1** and **Data2** are grouped and assigned to the **y-Axis**.
**Data3** is assigned to the **y2-Axis**.
When setting `bar.zerobased: false` it results in following chart:
![grafik](https://github.com/naver/billboard.js/assets/42860868/a93d585e-7d66-4e47-b578-f11089c0a826)

I would expect the y2-Axis to be also scaled like the y-Axis, as all values on this axis are positive too. 
I see no reason here to treat the y2-axis differently from the y-axis in this case.

Using `bar.zerobased: true` leads to following and correct result:
![grafik](https://github.com/naver/billboard.js/assets/42860868/d4b7b209-96d7-4771-a50b-558bde9d0c2b)

Removing **data2** from the chart and setting `bar.zerobased: false` results in this chart:
![grafik](https://github.com/naver/billboard.js/assets/42860868/522b8c6e-a4c7-4aa4-8814-7f564ab198fc)
Here I would also expect for both axes to be scaled like the Y-Axis in Screenshot 1 (the stacked bars)

`bar.zerobased: true` scales as expected: 
![grafik](https://github.com/naver/billboard.js/assets/42860868/c38f5374-5791-4ba9-b83c-9c5f21fcca29)

Having 2 grouped data rows on each axis results in both being scaled correctly:
![grafik](https://github.com/naver/billboard.js/assets/42860868/5dca3a6d-ba7e-4c1e-8fe5-83549cf7ffba)

Same with all 4 data rows being not grouped:
![grafik](https://github.com/naver/billboard.js/assets/42860868/8d705694-a442-4c44-8fb5-208a2db29845)

Overall, it seems like there is an inconsistency how the grouped data affects the scaling of the y-axes. It seems like you need at least 2 data rows per group for the scaling to work as expected.

Is the "wrong" behavior I described above the expected behavior, or can we classify this as a bug?

## Steps to check or reproduce
https://jsfiddle.net/michkami/a9uzyrs8/

**Comment by netil** [comment]
@michkami 
When there're 2 or more values aligned to single axis(y or y2), the domain range will set according it.

Let's say, given values are 300 and 500, in this case, min=300 and max=500.
But when single value (ex. 500) is given, it can determine min/max, because there's no data to determine or compare.

So, in this case min will be set as 0 and max 500.
https://github.com/naver/billboard.js/blob/master/src/ChartInternal/internals/domain.ts#L136

**Comment by michkami** [comment]
> @michkami When there're 2 or more values aligned to single axis(y or y2), the domain range will set according it.
> 
> Let's say, given values are 300 and 500, in this case, min=300 and max=500. But when single value (ex. 500) is given, it can determine min/max, because there's no data to determine or compare.
> 
> So, in this case min will be set as 0 and max 500. https://github.com/naver/billboard.js/blob/master/src/ChartInternal/internals/domain.ts#L136

I see, this makes sense.

Maybe not completely connected to the question above, but I found this line here https://github.com/naver/billboard.js/blob/487a1291dfe56ba90da3cbb52596a05cfa4ae2e8/src/ChartInternal/internals/domain.ts#L143 which cancels the zerobased setting when y min or max is set. But in case of `y.min: 0` or `y.max: 0` I would still expect the zerobased option put the zero on the x axis level.

Do you see any reason to not add an exception for the 0 values here?

In this case I would expect the zerobased option to still be working:
https://jsfiddle.net/michkami/6vr1z20f/

**Comment by netil** [comment]
> Do you see any reason to not add an exception for the 0 values here?

The conditional stayed from the start of the project and maintained as, mainly to not break any possible issue.
https://github.com/naver/billboard.js/blob/1.0.0/src/internals/domain.js#L156

**Comment by michkami** [comment]
> The conditional stayed from the start of the project and maintained as, mainly to not break any possible issue. https://github.com/naver/billboard.js/blob/1.0.0/src/internals/domain.js#L156

Yes, I saw this. My question was if there it would be possible to add 0 as an exception here in case `zerobased` is true, so that the y axis will start on the x axis level when min is set to 0. 

The current behaviour is what i would only expect if I set `zerobased: false` and then set the min value to 0.
If `zerobased: true` and I set the y axis min to 0, I would expect for it to start at the same level as the x axis.
I played around and saw, that if i pass `null` to y axis min, it would in this case work as expected, but the types only accept `number | undefined`.
